packages/headless/src/controllers/commerce/context/headless-context.ts

```diff
@@ -0,0 +1,212 @@
+import { Schema } from "@coveo/bueno";
+import {requiredNonEmptyString} from '../../../utils/validate-payload';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../controller/headless-controller';
+
+const optionsSchema = new Schema({
+  url: requiredNonEmptyString,
+});
+
+export interface User {
+  userId?: string;
+  email?: string;
+  userIp?: string;
+  userAgent?: string;
+}
+
+export interface View {
+  url: string;
+}
+
+export interface Product {
+  groupId?: string;
+  productId?: string;
+  sku?: string;
+}
+
+export interface CartItem {
+  product: Product;
+  quantity: number;
+}
+
+export interface ContextOptions {
+  trackingId?: string;
+  language?: string;
+  currency?: string;
+  // TODO: Does it make sense for the clientId to be part of the context state?
+  clientId?: string;
+  user?: User;
+  view?: Partial<View>;
+  cart?: CartItem[];
+  labels?: Record<string, string>
+}
+
+export interface ContextProps {
+  /**
+   * The initial options that should be applied to this `Context` controller.
+   */
+  options?: ContextOptions;
+}
+
+// TODO: Does this controller do too much? Should it expose sub-controllers which allow mutating
```

Comment:
I would probably create a distinct controller for the cart, because it's less "static" than the rest of the "global" commerce context. Otherwise, as I said in another comment, I believe `view` and `labels` (if we choose to expose it) should both be on the ProductListing controller.

Reply:
Created a cart controller! Great idea!

Left this comment unresolved while we figure out by which controller we handle the `view` (https://github.com/coveo/ui-kit/pull/3190/commits/9382bfa695c12e5b51aa4f16ec4f1b3e2acd1a80#r1330518824)